sdl-snowflake/src/main/scala/io/smartdatalake/workflow/dataframe/snowflake/SnowparkSubFeed.scala

```diff
@@ -0,0 +1,225 @@
+      case _ => SnowparkSubFeed(None, subFeed.dataObjectId, subFeed.partitionValues, subFeed.isDAGStart, subFeed.isSkipped)
+    }
+  }
+  override def subFeedType: universe.Type = typeOf[SnowparkSubFeed]
+  override def col(colName: String): SnowparkColumn = {
+    SnowparkColumn(functions.col(colName))
+  }
+  override def lit(value: Any): SnowparkColumn = {
+    SnowparkColumn(functions.lit(value))
+  }
+  override def max(column: GenericColumn): SnowparkColumn = {
+    column match {
+      case snowparkColumn: SnowparkColumn => SnowparkColumn(functions.max(snowparkColumn.inner))
+      case _ => throw new IllegalStateException(s"Unsupported subFeedType ${column.subFeedType.typeSymbol.name} in method max")
+    }
+  }
+  override def count(column: GenericColumn): SnowparkColumn = {
+    column match {
+      case snowparkColumn: SnowparkColumn => SnowparkColumn(functions.count(snowparkColumn.inner))
+      case _ => throw new IllegalStateException(s"Unsupported subFeedType ${column.subFeedType.typeSymbol.name} in method count")
+    }
+  }
+  override def explode(column: GenericColumn): SnowparkColumn = {
+    // TODO: check if this can be done by a udf?
+    throw new NotImplementedError("explode array is not implemented in Snowpark")
+  }
+  override def getEmptyDataFrame(schema: GenericSchema, dataObjectId: DataObjectId)(implicit context: ActionPipelineContext): SnowparkDataFrame = {
+    schema match {
+      case snowparkSchema: SnowparkSchema =>
+        val dataObject = context.instanceRegistry.get[SnowflakeTableDataObject](dataObjectId)
+        SnowparkDataFrame(dataObject.snowparkSession.createDataFrame(Seq(), snowparkSchema.inner))
+      case _ => throw new IllegalStateException(s"Can not create SnowparkDataFrame for ${schema.subFeedType.typeSymbol.name}")
+    }
+  }
+  override def getSubFeed(df: GenericDataFrame, dataObjectId: DataObjectId, partitionValues: Seq[PartitionValues])(implicit context: ActionPipelineContext): SnowparkSubFeed = {
+    df match {
+      case snowparkDf: SnowparkDataFrame => SnowparkSubFeed(Some(snowparkDf), dataObjectId, partitionValues)
+      case _ => throw new IllegalStateException(s"Can not create SnowparkSubFeed for ${df.subFeedType.typeSymbol.name}")
+    }
+  }
+  override def stringType: SnowparkDataType = SnowparkDataType(StringType)
+  override def arrayType(dataType: GenericDataType): SnowparkDataType = {
+    dataType match {
+      case snowparkDataType: SnowparkDataType => SnowparkDataType(ArrayType(snowparkDataType.inner))
+      case _ => throw new IllegalStateException(s"Can not create SnowparkDataType for ${dataType.subFeedType.typeSymbol.name}")
+    }
+  }
+  override def structType(fields: Map[String,GenericDataType]): SnowparkDataType = {
+    assert(fields.values.forall(_.isInstanceOf[SnowparkDataType]), s"Unsupported subFeedType(s) ${fields.values.filterNot(_.isInstanceOf[SparkDataType]).map(_.subFeedType.typeSymbol.name).toSeq.distinct.mkString(", ")} in method structType")
```

Comment:
same comment here as for
https://github.com/smart-data-lake/smart-data-lake/pull/486#discussion_r832001789

Reply:
Ok, created a helper method in DataFrameSubFeed